scipy/linalg/tests/test_batch.py

```diff
@@ -21,36 +29,75 @@ def get_nearly_hermitian(shape, dtype, atol, rng):
         return res
 
+    @pytest.fixture
+    def rng(self):
+        return np.random.default_rng(8342310302941288912051)
+
     @pytest.mark.parametrize('dtype', floating)
-    def test_expm_cond(self, dtype):
-        self.batch_test(linalg.expm_cond, dtype=dtype)
+    def test_expm_cond(self, dtype, rng):
+        A = rng.random((5, 3, 4, 4)).astype(dtype)
+        self.batch_test(linalg.expm_cond, A)
 
     @pytest.mark.parametrize('dtype', floating)
-    def test_issymmetric(self, dtype):
-        res = self.batch_test(linalg.issymmetric, dtype=dtype, kwargs=dict(atol=1e-3))
+    def test_issymmetric(self, dtype, rng):
+        A = get_nearly_hermitian((5, 3, 4, 4), dtype, 3e-4, rng)
+        res = self.batch_test(linalg.issymmetric, A, kwargs=dict(atol=1e-3))
         assert not np.all(res)  # ensure test is not trivial: not all True or False;
         assert np.any(res)      # also confirms that `atol` is passed to issymmetric
 
     @pytest.mark.parametrize('dtype', floating)
-    def test_ishermitian(self, dtype):
-        res = self.batch_test(linalg.issymmetric, dtype=dtype, kwargs=dict(atol=1e-3))
+    def test_ishermitian(self, dtype, rng):
+        A = get_nearly_hermitian((5, 3, 4, 4), dtype, 3e-4, rng)
+        res = self.batch_test(linalg.ishermitian, A, kwargs=dict(atol=1e-3))
         assert not np.all(res)  # ensure test is not trivial: not all True or False;
         assert np.any(res)      # also confirms that `atol` is passed to ishermitian
 
+    @pytest.mark.parametrize('dtype', floating)
+    def test_diagsvd(self, dtype, rng):
+        A = rng.random((5, 3, 4)).astype(dtype)
+        self.batch_test(linalg.diagsvd, A, args=(6, 4), core_dim=1)
+
+    @pytest.mark.parametrize('fun', [linalg.inv, linalg.sqrtm, linalg.signm,
+                                     linalg.sinm, linalg.cosm, linalg.tanhm,
+                                     linalg.sinhm, linalg.coshm, linalg.tanhm])
+    @pytest.mark.parametrize('dtype', floating)
+    def test_matmat(self, fun, dtype, rng):  # matrix in, matrix out
+        A = get_random((5, 3, 4, 4), dtype=dtype, rng=rng)
+        self.batch_test(fun, A)
+
+    @pytest.mark.parametrize('dtype', floating)
+    def test_null_space(self, dtype, rng):
+        A = get_random((5, 3, 4, 6), dtype=dtype, rng=rng)
+        self.batch_test(linalg.null_space, A)
+
+    @pytest.mark.parametrize('dtype', floating)
+    def test_funm(self, dtype, rng):
+        A = get_random((2, 4, 3, 3), dtype=dtype, rng=rng)
+        self.batch_test(linalg.funm, A, kwargs=dict(func=np.sin))
+
+    @pytest.mark.parametrize('dtype', floating)
+    def test_fractional_matrix_power(self, dtype, rng):
+        A = get_random((2, 4, 3, 3), dtype=dtype, rng=rng)
+        self.batch_test(linalg.fractional_matrix_power, A, args=(1.5,))
+
+    @pytest.mark.parametrize('dtype', floating)
+    def test_logm(self, dtype, rng):
+        with warnings.catch_warnings():  # "result not accurate"
```

Comment:
Looks like the ci failures are related to not using `numpy.testing.suppress_warnings`

Reply:
<sigh>

Reply:
Actually it turns out that increasing the diagonal elements was enough to avoid these warnings, too, so I was able to just remove the filter.